### PR TITLE
Sphinx doc

### PR DESCRIPTION
**Pull request by adriendelsalle**
Build the documentation using Sphinx and the `pydata` theme.

This first implementation is broadly inspired by [`Jupyter`](https://jupyter.readthedocs.io/en/latest/), [`pandas`](https://pandas.pydata.org/docs/index.html) and [`bokeh`](https://docs.bokeh.org/en/latest/index.html) docs.

Discussion
---
- At this time, a small part of the code is covered by docstrings
- *autodoc* has to be improved to display nicely classes/methods/attributes/etc. in `API reference`
- Configuration options to be detailed in `User Guide`

### CODE REVIEW

File: quetz/cli.py [hunk 1]
```diff
@@ -278,11 +278,11 @@ def start(path: str,
 
     Parameters
     ----------
-    path : str
+    path: str
```
Comment (benjamb) [inline quetz/cli.py:281]: I thought we'd agreed to keep the spaces before the `:`?

Comment (adriendelsalle) [reply quetz/cli.py:281]: Hi @benjamb 

You're right! I told you it is my own habit to write docstrings without whitespace-before-colon..

This only makes the built html having/not having thoses whitespaces, so we can just remove them everywhere if you prefer.

Comment (adriendelsalle) [reply quetz/cli.py]: I fixed the cli file. Thanks for having noticed that!
Let me know if you prefer to change the docstring style everywhere.

Comment (benjamb) [reply quetz/cli.py]: I'm not that bothered myself tbh, though I agree with @wolfv that consistency is what's important.

Comment (adriendelsalle) [reply quetz/cli.py]: I also agree!

**Comment by wolfv** [comment]
we've added docs now (using sphinx: #127), and they are available at https://quetz.readthedocs.io/ )

**Comment by adriendelsalle** [comment]
Cool!
Sorry this draft PR got stale, I didn't took time to request a review.

**Comment by wolfv** [comment]
no problem at all!

**Pull request closed without merging**